index.bs

```diff
@@ -493,15 +493,21 @@ This Section defines a number of new set of <a>Constrainable Properties</a> for
   <dd>This reflects the supported range of <a>contrast</a>. Values are numeric.  Increasing values indicate increasing contrast.</dd>
 
   <dt><dfn dict-member for="MediaTrackCapabilities"><code>pan</code></dfn></dt>
-  <dd>This reflects the <a>pan</a> value range supported by the UA and by the track.
+  <dd>
+    This reflects the <a>pan</a> value range supported by the UA and by the track.
 
-  If the track has been created without <a>requesting permission to use</a> (as defined in [[!permissions]]) a PermissionDescriptor with its name member set to {{PermissionName/camera}} and its {{CameraDevicePermissionDescriptor/panTiltZoom}} member set to true or if that permission request is denied, the track does not support <a>pan</a>.
-  In that case the UA MUST NOT expose the <a>pan</a> value range.
+    If the track has been created without <a>requesting permission to use</a> (as defined in [[!permissions]]) a PermissionDescriptor with its name member set to {{PermissionName/camera}} and its {{CameraDevicePermissionDescriptor/panTiltZoom}} member set to true or if that permission request is denied, the track does not support <a>pan</a>.
```

Comment:
This sentence seems to suggest the relevant permissions are the ones granted at the **_time of creation_** of the track, which seems wrong.

If the site sees `pan: {}`, decides to requests more permission with `applyConstraints({pan: true})`, and the user approves, then it should see `pan: {min: -648000, max: +648000}` from that point on.

To do that requires that we look at the _current_ permissions, i.e. from a fresh permission descriptor. Can you add that?

Reply:
> This sentence seems to suggest the relevant permissions are the ones granted at the **_time of creation_** of the track, which seems wrong.

That's right, and it isn't wrong. Track capabilities are effectively constant and do not change during the life-time of the track.

> If the site sees `pan: {}`, decides to request more permission with `track.applyConstraints({pan: true})`, and the user approves, then it should see `pan: {min: -648000, max: +648000}` from that point on.

If the site sees `pan: {}`, it can decide to request more permission with `navigator.mediaDevices.getUserMedia({video: {deviceId: {exact: track.getSettings().deviceId}, pan: true})`, and if the user approves, then it should see `pan: {min: -648000, max: +648000}` from that point on.

If the site sees `pan: {}` (without min and max), the track has been created without `pan: true` (or tilt or zoom) thus the track does not support PTZ. Therefore, `track.applyConstraints({pan: true})` does nothing (because the ideal constraint cannot be fulfilled) and `track.applyConstraints({pan: {exact: 0}})` throws `OverconstrainedError` (because the required constraint cannot be fulfilled).

Reply:
> That's right, and it isn't wrong. Track capabilities are effectively constant and do not change during the life-time of the track.

This is the current consensus that was discussed during the interim, I believe for simplicity.
I am not opposed to revisiting it but this should be tackled as a separate issue.

Reply:
If that is revisited, the main spec should probably also be changed because it specifies that capabilities are effectively constant.

Reply:
I agree with @youennf here. @eehakkin mediacapture-main [says](https://w3c.github.io/mediacapture-main/getusermedia.html#dfn-capabilities): _"Source capabilities are effectively constant. Applications should be able to depend on a specific source having the same capabilities for any browsing session."_

I think we're fine because the underlying (source) capabilities aren't changing, we're just un-filtering them. There's [precedence](https://w3c.github.io/mediacapture-main/getusermedia.html#dom-inputdeviceinfo-getcapabilities) for that.

The premise of this PR is to detect the availability of a PTZ device, so the site can act on it.